### ISSUE TITLE
forbid-prop-types: depends on how `objectOf` is imported

### ISSUE THREAD

**Issue by thany** [yannickcr/eslint-plugin-react#2662]
Consider the following rule:
```json
"react/forbid-prop-types": [ "error", { "forbid": [ "any", "object", "objectOf", "array" ] } ],
```

This would yield an error, which is correct:
```js
import PropTypes from 'prop-types';

AwesomeComponent.propTypes = {
  foo: PropTypes.objectOf(PropTypes.any)
};
```

This would also yield an error, which is correct:
```js
import { any } from 'prop-types';

AwesomeComponent.propTypes = {
  foo: any
};
```

This *should* yield an error, but doesn't:
```js
import { objectOf, any } from 'prop-types';

AwesomeComponent.propTypes = {
  foo: objectOf(any)
};
```
This is slightly related to #2657 where `any` should be blocked in the above example, but isn't. But the bug I'm describing here, is that `objectOf` *only* yields an error if it's imported as part of a default import, where it *doesn't* yield an error when imported as a destructured import. Strangely though, `any` is not affected by this phenomenon, only `objectOf` is.

I'm on eslint-plugin-react 7.19.0, as part of react-scripts 3.4.1.

**Comment by ljharb** [comment]
Please confirm with v7.20.0, but otherwise this is definitely a bug.

**Comment by thany** [comment]
Just tried, and this seems to happen in 7.20.0 as well.

**Comment by ljharb** [comment]
I've gotten a failing test written, but haven't been able to quickly figure out how to fix it. A PR would be most appreciated.

**Comment by ajkovar** [comment]
Nice test demo.  I think I might have a fix in mind for it this nobody minds me taking a crack at it.

**Comment by ljharb** [comment]
Please do!
